src/lib/db/feature-strategy-store.ts

```diff
@@ -361,10 +418,25 @@ class FeatureStrategiesStore implements IFeatureStrategiesStore {
                         environments: [
                             FeatureStrategiesStore.getEnvironment(r),
                         ],
+                        tags: [],
                     };
+                    if (this.isUnseenTag(acc[r.feature_name], r)) {
+                        acc[r.feature_name].tags.push(
+                            FeatureStrategiesStore.getTag(r),
+                        );
+                    }
                 }
                 return acc;
             }, {});
+
+            if (!this.flagResolver.isEnabled('toggleTagFiltering')) {
```

Comment:
I do not understand why we need this? It would not be added if we do not do the left join? Cant we just add the tags property the first time we try to push to the list?